bolt/lib/Profile/StaleProfileMatching.cpp

```diff
@@ -555,6 +574,10 @@ size_t matchWeightsByHashes(
                            ProbeMap.lower_bound(FuncAddr + BlockRange.second));
       for (const auto &[_, Probes] : BlockProbes) {
         for (const MCDecodedPseudoProbe &Probe : Probes) {
+          if (Probe.getInlineTreeNode()->hasInlineSite())
```

Comment:
What do we prune with this check? Don't we discard valid probes belonging to the current function?

Reply:
This pruning resulted in a tangible increase in 1:1 mappings btw profile and binary pseudo probes - the PseudoProbeDecoder::ProbeMap interface not only contains the probe attached to some block b, but contains the block probes attached to block b inlined in other functions.